internal/server/middleware.go

```diff
@@ -305,7 +305,7 @@ func reqBearerToken(c *gin.Context, baseDomain string) (string, error) {
 		*/
 		cookie, err := getCookie(c.Request, cookieSignupName)
 		if err == nil {
-			exchangeSignupCookieForSession(c, baseDomain)
+			exchangeSignupCookieForSession(c, opts)
```

Comment:
Not a blocker, but I wonder if we could remove the duplicate lines:

```
cookie, err := getCookie(c.Request, cookieSignupName)
```

There's one just above here, and another in `exchangeSignupCookieForSession`.

I suspect either `exchangeSignupCookieForSession` could return a `handled bool` to indicate to the caller how to proceed. Or we can look up the cookie here and pass it into `exchangeSignupCookieForSession`

Reply:
Made this change, pretty clean.